include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -301,8 +301,6 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
                 auto total_weight = std::accumulate(weight_vector.begin(), weight_vector.end(), 0);
 
                 BOOST_ASSERT(weight_vector.size() == id_vector.size());
-                // ed.distance should be total_weight + penalties (turn, stop, etc)
-                BOOST_ASSERT(ed.distance >= total_weight);
```

Comment:
@TheMarex yes, that's exactly the case this was removed to support.

The situation where this might arise would be where segment speeds are being calculated using GPS probe data.  At an intersection where the majority of the traffic makes a turn, the average speed will be primarily that of turning traffic.  However, if the intersection also allows unimpeded straight-through movements, a negative "turn" value can be used to correct traversal time for traffic that proceeds straight and unimpeded.  This type of situation is reasonably common on the outskirts of urban areas across the US, example: http://www.openstreetmap.org/#map=18/38.79196/-121.35847

Really though, it would apply in any situation where an average measurement does not represent the fastest traffic through an intersection, and a "speedup" needs to be applied for some movements.